src/SortingTable/SortingTable.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from "react";
+import { Table, TableProps } from "../Table";
+
+type SortingTableProps = TableProps & {
+  initialSortColumn: string;
+};
+
+const numericAlphabeticalSort = (a, b, numeric) =>
+  String(a).localeCompare(b, undefined, { numeric, sensitivity: "base" });
+
+const applySort = (rows, sortColumn, columns) =>
+  rows.sort((a, b) => {
+    const column = columns.find((col) => col.dataKey === sortColumn);
+    const { numeric } = column;
+
+    return numericAlphabeticalSort(a[sortColumn], b[sortColumn], numeric);
+  });
+
+const sortRows = (rows, columns, sortState) => {
+  const sortedRows = applySort(rows, sortState.sortColumn, columns);
+
+  return sortState.ascending ? sortedRows : sortedRows.reverse();
+};
+
+const SortingTable = ({
+  columns: incomingColumns,
+  rows: incomingRows,
+  initialSortColumn,
+  ...props
+}: SortingTableProps) => {
```

Comment:
Do you mind changing this to use `React.FC<T>`

```
const SortingTable: React.FC<SortingTableProps> = ({
  columns: incomingColumns,
  rows: incomingRows,
  initialSortColumn,
  ...props
}) => {
```

Reply:
This must be a typescript thing?

Reply:
Done